tsdb/index/index.go

```diff
@@ -1849,10 +1854,11 @@ func (s stringListIter) Err() error { return nil }
 // by them if there's demand.
 type Decoder struct {
 	LookupSymbol func(context.Context, uint32) (string, error)
+	Postings     PostingsDecoder
```

Comment:
Nice :+1:

Reply:
Can you explain what `Decoder` is for?  I don't think I looked at it before, but before this PR it was a wrapper for a function pointer, which was exported.
The only use in Prometheus was in `Reader`, as an unexported data member.

This PR gives us both `Decoder` and `PostingsDecoder` in this file, which also leaves me confused.  There is a `PostingsEncoder` but no `Encoder`.  What's going on?

I can't see where this newly-added member `Postings` (why that name?) is used in this PR.

Reply:
Previously the function that decoded postings was called Postings() so to make fewer changes, this field is named the same way — this way the callers don't need to be changed. 

This decoder is a container of methods for decoding data in index files and allows overriding them. This change compared to the postings encoding part is a bit more involved because we the compactor might compact blocks with different postings formats in one group hence there's a need to dynamically do that.

Reply:
Renamed the field `Postings` to `DecodePostings`. @bboreham